### PR TITLE
Add basic sidebearing editing

### PR DESCRIPTION
**Pull request by cmyr**
This adds a new floating panel via which the user can
edit the left and right sidebearings.

This needs some polish, particularly around how we draw
the little placeholder glyph; ultimately I'm not sure
what the UX should really be here.

### CODE REVIEW

**Comment by raphlinus** [comment]
So far just playing with it, not looking at the code. It's not behaving quite as I expected, as applying a delta to both LSB and RSB should have that delta applied to the advance width of the glyph as well. Other than that it seems like a nice feature.

**Comment by cmyr** [comment]
can you clarify the behaviour you'd like?

Currently an edit of a sidebearing is just translating the entire glyph on the x-axis. You would like it to modify the advance width? Or only in the case where both are edited?

If you want this to be the case when both are edited that will be slightly tricky, because the edit ends and is applied whenever the textbox loses focus. I do want to also make the advance width editable from this panel, though..

**Comment by raphlinus** [comment]
> can you clarify the behaviour you'd like?

Yes. Let's say lsb is 10, rsb is 15, and adv width is 500, just for round numbers. Edit lsb to 20: translate glyph +10 (which it does now) and change adv width to 500.

Edit rsb to 20: *don't translate* glyph (ie revert that behavior) and change adv width to 505. Ie an edit of rsb is equivalent to an edit of adv width.

**Comment by cmyr** [comment]
@raphlinus this is updated and should have the behaviour you were looking for.

I also need to go make it so that textboxes can specify their alignment, a bunch of stuff looks bad right now.

File: src/widgets/glyph_pane.rs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                ),
+        )
+        .with_child(
+            EditableLabel::parse()
+                .with_font(theme::UI_DETAIL_FONT)
+                .lens(EditorState::detail_glyph.then(GlyphDetail::advance))
+                .fix_width(40.0),
+        )
+        .padding(4.0)
+        .debug_paint_layout()
```
Comment (raphlinus) [inline src/widgets/glyph_pane.rs:95]: I think maybe you didn't mean to check this in.

**Review by raphlinus — APPROVED** [review]
Works great now, and the code looks reasonable. Only nit is the debug layout painting.

**Pull request merged**